week3/createContentTrainingData.py

```diff
@@ -52,7 +59,8 @@ def transform_name(product_name):
                     child.find('categoryPath')[len(child.find('categoryPath')) - 1][0].text is not None):
                       # Choose last element in categoryPath as the leaf categoryId
                       cat = child.find('categoryPath')[len(child.find('categoryPath')) - 1][0].text
+                      product_counter[cat]=product_counter.get(cat,0)+1
```

Comment:
That's a clever way to use a default to get this in one line.  I sometimes use `defaultdict` for such things:
```
from collections import defaultdict
product_counter = defaultdict(lambda: 0)
...
product_counter[cat] += 1
```
That involves a little more setup than you use here, though.